table/block_based/partitioned_filter_block.cc

```diff
@@ -477,17 +475,18 @@ void PartitionedFilterBlockReader::CacheDependencies(const ReadOptions& ro,
         prefetch_buffer.get(), ro, handle, UncompressionDict::GetEmptyDict(),
         &block, BlockType::kFilter, nullptr /* get_context */, &lookup_context,
         nullptr /* contents */);
-
-    assert(s.ok() || block.GetValue() == nullptr);
```

Comment:
I think this assertion is still valid?

Reply:
yes it is. I recently got to know that assertions are excluded in optimized builds.